src/macros.h

```diff
@@ -22,4 +22,10 @@
 
 #define except(x) if (!(x)) { wlc_log(WLC_LOG_ERROR, "assertion failed: %s", #x); abort(); }
 
+#ifdef HAVE_CLEARENV
+#define CLEARENV() if (clearenv() != 0) { die("Failed to clear environment"); }
+#else
+#define CLEARENV() extern char **environ; environ[0] = NULL;
+#endif
```

Comment:
I would prefer static inline functions instead of macros, maybe in internal.h as you suggested.